src/v/kafka/server/group.h

```diff
@@ -202,6 +202,7 @@ class group final : public ss::enable_lw_shared_from_this<group> {
       kafka::group_id id,
       group_state s,
       config::configuration& conf,
+      ss::lw_shared_ptr<ss::basic_rwlock<>> catchup_lock,
```

Comment:
curious why a shared_ptr instead of a reference? I think group doesn't technically own the lock (no shared ownership, manager owns it) and a group is destroyed before lock is destroyed, so reference seems more apt.

Reply:
personal preference, references don't mix seamlessly with co-routines and it's easier to shot in a leg (e.g. when stack is destroyed before the next co-routine gets control flow) so opted for lw_shared_ptr here